packages/binding-mbus/src/mbus-connection.ts

```diff
@@ -311,7 +316,11 @@ export class PropertyOperation {
         const resp = new Content("application/json", Readable.from(JSON.stringify(payload)));
 
         // resolve the Promise given to the invoking script
-        this.resolve(resp);
+        if (this.resolve) {
+            this.resolve(resp);
+        } else {
+            warn("resolve undefined");
+        }
```

Comment:
In Modbus we are throwing an error: 
```
   failed(reason: Error): void {
        warn(`Operation failed: ${reason}`);
        if (!this.reject) {
            throw new Error("Function 'failed' was invoked before executing the Modbus operation");
        }
        // reject the Promise given to the invoking script
        this.reject(reason);
    }
```
What do you think?

Reply:
Alignment makes sense